File: python-algorithm/ds_3_hash/L594.py
# -*- coding: utf-8 -*-

class Solution:
    def findLHS(self, nums: List[int]) -> int:
        d = {}
        res = 0
        for i in range(len(nums)):
            if nums[i] not in d:
                d[nums[i]] = 1
            else:
                d[nums[i]] += 1
        for i in range(len(nums)):
            if nums[i]-1 in d:
                res = max(res, d[nums[i]]+d[nums[i]-1])
            if nums[i]+1 in d:
                res = max(res, d[nums[i]]+d[nums[i]+1])
        return res

File: python-algorithm/ds_3_hash/L594.cpp

#include <iostream>
#include <map>
#include <vector>
using namespace std;
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int res = 0;
        //使用map 因为有有序的需求
        map<int,int> mp;
        for(auto n:nums)
            ++mp[n];
        for(auto p:mp){
            if(mp.find(p.first+1) != mp.end())
                res = max(res, p.second+mp[p.first+1]);
        }
        return res;
    }
};